paasta_tools/kubernetes_tools.py

```diff
@@ -2293,12 +2310,24 @@ def get_node_affinity(self) -> Optional[V1NodeAffinity]:
             allowlist=self.get_deploy_whitelist(),
             denylist=self.get_deploy_blacklist(),
         )
+        node_selectors = self.config_dict.get("node_selectors", {})
         requirements.extend(
             raw_selectors_to_requirements(
-                raw_selectors=self.config_dict.get("node_selectors", {}),
+                raw_selectors=node_selectors,
             )
         )
 
+        # PAASTA-18198: To improve AZ balance with Karpenter, we temporarily allow specifying zone affinities per pool
+        if pool_node_affinities:
+            current_pool_node_affinities = pool_node_affinities[self.get_pool()]
```

Comment:
is pool_node_affinities guaranteed to always contain the current pool?

Reply:
thanks, added a check